### PR TITLE
WA-150: DesignSafe Tapis v3: LS-DYNA

### PR DESCRIPTION
**Pull request by asimregmi**
- [WA-150](https://tacc-main.atlassian.net/browse/WA-150)

Total 4 apps:

[LS-DYNA ( Frontera)](https://dev.cep.tacc.utexas.edu/workbench/applications/LS-Dyna) 

[LS-DYNA (Stampede3)](https://dev.cep.tacc.utexas.edu/workbench/applications/LS-Dyna-stampede3)

[LS-PrePost (Frontera)](https://dev.cep.tacc.utexas.edu/workbench/applications/ls-pre-post?appVersion=4.10.3) 

[LS-PrePost (Stampede3)](https://dev.cep.tacc.utexas.edu/workbench/applications/lsprepost-stampede3?appVersion=4.10.6)

### CODE REVIEW

File: applications/LS-Dyna-stampede3/app.json [hunk 1]
```diff
@@ -0,0 +1,124 @@
+{
+    "id": "LS-Dyna-stampede3",
+    "version": "2024R1",
```
Comment (rstijerina) [inline applications/LS-Dyna-stampede3/app.json:3]: Just curious, where is the version referenced from?

File: applications/LS-Dyna-stampede3/app.json [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        "subscriptions": [],
+        "tags": []
+    },
+    "tags": [
+        "portalName: DesignSafe"
+    ],
+    "notes": {
+        "icon": "LS-Dyna",
+        "label": "LS-Dyna (Stampede3)",
+        "helpUrl": "https://docs.tacc.utexas.edu/software/ansys/",
```
Comment (rstijerina) [inline applications/LS-Dyna-stampede3/app.json:119]:
```suggestion
        "helpUrl": "https://www.designsafe-ci.org/user-guide/tools/simulation/lsdyna/",
```

File: applications/LS-Dyna-stampede3/app.json [hunk 3]
```diff
@@ -0,0 +1,124 @@
+{
+    "id": "LS-Dyna-stampede3",
```
Comment (rstijerina) [inline applications/LS-Dyna-stampede3/app.json:2]: can we keep the `id` lowercase?

File: applications/LS-Dyna-stampede3/app.json [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+        "archiveOnAppError": true,
+        "isMpi": false,
+        "mpiCmd": null,
+        "cmdPrefix": null,
+        "parameterSet": {
+            "appArgs": [
+                {
+                    "name": "Input file",
+                    "description": "Please enter the input file in .txt format. Add one .k case file per line if there are multiple case files.",
+                    "arg": "myfilename.txt",
```
Comment (rstijerina) [inline applications/LS-Dyna-stampede3/app.json:35]: Is this a valid default arg? If not, you can leave blank.

File: applications/LS-Dyna-stampede3/app.json [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+                    }
+                }
+            ],
+            "containerArgs": [],
+            "schedulerOptions": [
+                {
+                    "name": "TACC Scheduler Profile",
+                    "description": "Scheduler profile for HPC clusters at TACC",
+                    "inputMode": "FIXED",
+                    "arg": "--tapis-profile ls-dyna-2023R2",
```
Comment (rstijerina) [inline applications/LS-Dyna-stampede3/app.json:51]: The profile version differs from the app version. Is that intentional?

File: applications/LS-Dyna-stampede3/app.json [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+                "includeLaunchFiles": true
+            }
+        },
+        "fileInputs": [
+            {
+                "name": "Input Directory",
+                "description": "The directory containing the files that you want to work on. This directory and its files will be copied to where your GiD session runs. Click the 'Select Input' button and then select the directory you want to work from.",
+                "inputMode": "REQUIRED",
+                "autoMountLocal": true,
+                "sourceUrl": "tapis://cloud.data/corral/tacc/aci/CEP/community/app_examples/ls-dyna",
```
Comment (rstijerina) [inline applications/LS-Dyna-stampede3/app.json:101]: Let's leave the `sourceUrl` blank, and instead add the link to the description.

File: applications/lsprepost-stampede3/app.json [hunk 7]
```diff
@@ -0,0 +1,117 @@
+{
+    "id": "lsprepost-stampede3",
```
Comment (rstijerina) [inline applications/lsprepost-stampede3/app.json:2]: Can we change this to `ls-pre-post-stampede3`?

File: applications/LS-Dyna-stampede3/app.json (hunk 1 above)
Comment (asimregmi) [reply applications/LS-Dyna-stampede3/app.json:3]: It uses the ANSYS 2024R1 build

File: applications/LS-Dyna-stampede3/app.json (hunk 2 above)
Comment (asimregmi) [reply applications/LS-Dyna-stampede3/app.json:119]: I'll go ahead and update this URL. However, just a heads up, the information at [helpUrl](https://www.designsafe-ci.org/user-guide/tools/simulation/lsdyna/) seems to be a bit behind/outdated.

File: applications/LS-Dyna-stampede3/app.json (hunk 4 above)
Comment (asimregmi) [reply applications/LS-Dyna-stampede3/app.json:35]: This is valid. The case files name are included in the .txt file.

File: applications/LS-Dyna-stampede3/app.json (hunk 5 above)
Comment (asimregmi) [reply applications/LS-Dyna-stampede3/app.json:51]: Updated the profile name for both frontera and stampede3.

File: applications/LS-Dyna-stampede3/app.json [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+                    }
+                }
+            ],
+            "containerArgs": [],
+            "schedulerOptions": [
+                {
+                    "name": "TACC Scheduler Profile",
+                    "description": "Scheduler profile for HPC clusters at TACC",
+                    "inputMode": "FIXED",
+                    "arg": "--tapis-profile ls-dyna-stampede3",
```
Comment (rstijerina) [inline applications/LS-Dyna-stampede3/app.json:51]: Ah, sorry. Can we still include the version in the profile name?

Comment (asimregmi) [reply applications/LS-Dyna-stampede3/app.json]: Sorry, misunderstood your initial comment. Fixed it with this commit.

**Comment by asimregmi** [comment]
fixed app icon to LS-DYNA

**Pull request merged**